pkg/tagger/collectors/kubernetes_main.go

```diff
@@ -47,20 +47,18 @@ func (c *KubeMetadataCollector) Detect(out chan<- []*TagInfo) (CollectionMode, e
 		return NoCollection, err
 	}
 	// if no DCA or can't communicate with the DCA run the local service mapper.
-	if config.Datadog.GetBool("cluster_agent") {
+	if config.Datadog.GetBool("cluster_agent.enabled") {
 		c.dcaClient, errDCA = clusteragent.GetClusterAgentClient()
 		if errDCA != nil {
 			log.Errorf("Could not initialise the communication with the DCA, falling back to local service mapping: %s", errDCA.Error())
 		}
 	}
-	if !config.Datadog.GetBool("cluster_agent") || errDCA != nil {
+	if !config.Datadog.GetBool("cluster_agent.enabled") || errDCA != nil {
 		c.apiClient, err = apiserver.GetAPIClient()
 		if err != nil {
 			return NoCollection, err
 		}
-		c.apiClient.StartMetadataMapping()
```

Comment:
if you remove this call you can remove the whole block

Reply:
I think it is legit to leave the rest of the block as if we can't talk to the API Server, we don't want to start the `kube-metadata-collector` (so the Detect will fail)